graphql-server/src/schemas/neo4j.ts

```diff
@@ -336,109 +342,30 @@ function buildResolvers(
-          selectionSet: `{
-            datasource
-            revisable
-            sampleCategory
-            sampleClass
-            smileSampleId
-            hasMetadataSampleMetadata {
-              additionalProperties
-              baitSet
-              cfDNA2dBarcode
-              cmoInfoIgoId
-              cmoPatientId
-              cmoSampleIdFields
-              cmoSampleName
-              collectionYear
-              genePanel
-              igoComplete
-              igoRequestId
-              importDate
-              investigatorSampleId
-              libraries
-              oncotreeCode
-              preservation
-              primaryId
-              qcReports
-              sampleClass
-              sampleName
-              sampleOrigin
-              sampleType
-              sex
-              species
-              tissueLocation
-              tubeId
-              tumorOrNormal
-              hasStatusStatuses {
-                validationReport
-                validationStatus
-              }
-            }
-            requestsHasSample {
-              igoRequestId
-              igoProjectId
-              genePanel
-              dataAnalystName
-              dataAnalystEmail
-              dataAccessEmails
-              bicAnalysis
-              investigatorEmail
-              investigatorName
-              isCmoRequest
-              labHeadEmail
-              labHeadName
-              libraryType
-              otherContactEmails
-              piEmail
-              projectManagerName
-              qcAccessEmails
-              smileRequestId
-            }
-            patientsHasSample {
-              smilePatientId
-              patientAliasesIsAlias {
-                namespace
-                value
-              }
-            }
-            cohortsHasCohortSample {
-              cohortId
-              hasCohortCompleteCohortCompletes {
-                date
-              }
-            }
-            hasTempoTempos {
-              smileTempoId
-              billed
-              billedBy
-              costCenter
-              custodianInformation
-              accessLevel
-              hasEventBamCompletes {
-                date
-                status
-              }
-              hasEventMafCompletes {
-                date
-                normalPrimaryId
-                status
-              }
-              hasEventQcCompletes {
-                date
-                reason
-                result
-                status
-              }
-            }
-          }`,
-        });
-
-        return samples;
+      async samples(
+        _source: undefined,
+        args: any,
+        { oncotreeCache }: ApolloServerContext
+      ) {
+        let customWhere = includeCancerTypeFieldsInSearch(
+          args.where,
+          oncotreeCache
+        );
+        return await querySamplesList(ogm, customWhere, args.options);
+      },
```

Comment:
Before querying the database, we will modify the `where`/filter to include oncotree codes corresponding to the `ct`/`ctDetailed` values from users' search input.

Moved the database call into the named function `querySamplesList` because we have to use it again for the `samplesConnection` resolver below.